### PR TITLE
Added tests

### PR DESCRIPTION
**Pull request by l3l0**
Added test for FacebookAuthenticationEntryPoint
Added test for FacebookListener
Extended FOSFacebookExtensionTest
Added dependency injection configuration test
Added test for facebook security factory

Added ignore test converage for FOSFacebookBundle which is empty

### CODE REVIEW

File: FOSFacebookBundle.php [hunk 1]
```diff
@@ -13,6 +13,9 @@ namespace FOS\FacebookBundle;
 
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
+/**
+ * @codeCoverageIgnore 
+ */
```
Comment (stof) [inline FOSFacebookBundle.php]: I don't see the value of adding it. I'm not a fan of adding annotations specific to the tests in the code.

File: Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    private function facebookFactoryCreate($config = array())
+    {
+        $container = $this->getMock('Symfony\Component\DependencyInjection\ContainerBuilder', array('setDefinition', 'addArgument', 'replaceArgument'));
+        $container->expects($this->any())
+            ->method('replaceArgument')
+            ->will($this->returnValue($container));
+        $container->expects($this->any())
+            ->method('addArgument')
+            ->will($this->returnValue($container));
```
Comment (stof) [inline Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php]: This is wrong. `replaceArgument` and `addArgument` are not methods of the ContainerBuilder

File: Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        $container->expects($this->any())
+            ->method('setDefinition')
+            ->will($this->returnValue($container));
```
Comment (stof) [inline Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php]: this is wrong. The return value is the definition, not the container.

File: Tests/Security/EntryPoint/FacebookAuthenticationEntryPointTest.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        $response = $facebookAuthentication->start($requestMock);
+
+        $this->assertInstanceOf('Symfony\Component\HttpFoundation\RedirectResponse', $response, 'RedirectResponse is returned');
+        $this->assertEquals($response->headers->get('location'), 'http://localhost/facebook-redirect/index', 'RedirectResponse has defined expected location');
+    }
+
+    /**
+     * @covers FOS\FacebookBundle\Security\EntryPoint\FacebookAuthenticationEntryPoint::start
+     */
+    public function testThatResponseWithJsRedirectionToFacebookLoginUrlIsCreatedWhenAppAndServerUrlsAreDefined()
```
Comment (stof) [inline Tests/Security/EntryPoint/FacebookAuthenticationEntryPointTest.php]: isn't it possible to make the method name even more verbose ? :)

Comment (lsmith77) [inline Tests/Security/EntryPoint/FacebookAuthenticationEntryPointTest.php]: :)

File: Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php (hunk 2 above)
Comment (l3l0) [inline Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php]: Thanks, I didn't notice that. I will fix it today

File: Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php (hunk 3 above)
Comment (l3l0) [inline Tests/DependencyInjection/Security/Factory/FacebookFactoryTest.php]: I will fix it.

File: Tests/Security/EntryPoint/FacebookAuthenticationEntryPointTest.php (hunk 4 above)
Comment (l3l0) [inline Tests/Security/EntryPoint/FacebookAuthenticationEntryPointTest.php]: I'll try to be more concise ;)

File: FOSFacebookBundle.php (hunk 1 above)
Comment (l3l0) [inline FOSFacebookBundle.php]: I think that there is no point to coverage empty class. 
If you prefer I can do that via phpunit.xml.dist file.

I noticed that codeCoverageIgnore annotations are used in code:
https://github.com/FriendsOfSymfony/FOSFacebookBundle/blob/master/DependencyInjection/FOSFacebookExtension.php#L49
https://github.com/FriendsOfSymfony/FOSFacebookBundle/blob/master/DependencyInjection/FOSFacebookExtension.php#L57
https://github.com/FriendsOfSymfony/FOSFacebookBundle/blob/master/DependencyInjection/FOSFacebookExtension.php#L65
so I used there annotation too.

Comment (stof) [inline FOSFacebookBundle.php]: There is indeed no point to write tests for them, but there is no point either to add an annotation for it IMO. This simply makes you think you have 100% coverage which is wrong (even if you don't need to reach 100%)

The annotations in the DI extension should IMO be removed, especially as `loadDefaults` will be covered by a test testing `load` as it is used in it.

And the methods about XML should be removed IMO. We already validate the config anyway so adding an XSD is useless (and is not used as most people don't use XML for their app config but YAML). Thus the XSD is outdated, thus breaking the use of XML.

Comment (l3l0) [inline FOSFacebookBundle.php]: Ok then I will remove codeCoverageIgnore annotation from that class.

**Comment by lsmith77** [comment]
ok .. let us know when you had time do to the changes.

**Comment by l3l0** [comment]
I done changes - the last two commits

**Comment by lsmith77** [comment]
think you forgot to fix the super long method name ..

**Comment by l3l0** [comment]
Fixed.

**Comment by lsmith77** [comment]
thx!

merged ..

**Pull request merged**